sudo.cpp

```diff
@@ -169,30 +165,42 @@ int Sudo::main()
 
 void Sudo::child()
 {
-    int params_cnt = 2 //1. su/sudo & last nullptr
-        + 1 //-c for su | -E for sudo
-        + mArgs.size();
+    int params_cnt = 3 //1. su/sudo & "shell command" & last nullptr
+        + (BACK_SU == mBackend ? 1 : 3); //-c for su | -E /bin/sh -c for sudo
     std::unique_ptr<char const *[]> params{new char const *[params_cnt]};
     const char ** param_arg = params.get() + 1;
 
     std::string program;
     if (BACK_SU == mBackend)
     {
         program = su_prog.toStdString();
-        *(param_arg++) = "-c"; //run command
     } else
     {
         program = sudo_prog.toStdString();
         *(param_arg++) = "-E"; //preserve environment
+        *(param_arg++) = "/bin/sh";
     }
+    *(param_arg++) = "-c"; //run command
 
     params[0] = program.c_str();
 
-    std::vector<std::string> arguments;
-    for (const auto & a : mArgs)
-        arguments.push_back(a.toStdString());
-    for (const auto & a : arguments)
-        *(param_arg++) = a.c_str();
+    // Note: we force the su/sudo to communicate with us in the simplest
+    // locale and then set the locale back to previous for the command
```

Comment:
This doesn't flow right to a native English speaker; this would be better:
```
// Note: we force the su/sudo command to communicate with us in the simplest locale, and then set the locale back for the command
```
(ofc with line breaks)